t/t4129-apply-samemode.sh

```diff
@@ -100,5 +100,21 @@ test_expect_success POSIXPERM 'do not use core.sharedRepository for working tree
 		test_cmp d_mode.expected d_mode.actual
 	)
 '
+test_expect_success FILEMODE 'ensure git apply respects core.fileMode' '
+  test_tick && git commit -m initial --allow-empty &&
```

Comment:
We don't really need the initial commit here, do we? Also, please indent with tabs instead of 2 spaces.

Finally, the Git maintainer prefers these tests to be introduced together with the patch that fixes the problem. So that would be another `squash` in `git rebase -i`.